PostHog/Replay/PostHogReplayIntegration.swift

```diff
@@ -386,6 +391,22 @@
             return wireframe
         }
 
+        /// Check if window's root view controller is animating a transition
+        private func isAnimatingTransition(_ window: UIWindow) -> Bool {
+            guard let viewController = window.rootViewController else { return false }
+
+            return isAnimatingTransition(viewController)
+        }
+
+        // Recursively check if a view controller (or any of its children) is currently animating a transition
+        // During a transition, a UIViewControllerTransitionCoordinator is assigned to a UIViewController which controls the overall transition
+        private func isAnimatingTransition(_ viewController: UIViewController) -> Bool {
+            if let isAnimated = viewController.transitionCoordinator?.isAnimated {
+                return isAnimated
+            }
+            return viewController.children.contains(where: isAnimatingTransition)
+        }
```

Comment:
I have two questions here: Since this loops all its children, can it be slow? Should we introduce a maximum depth?
If the view is deeply nested, it gives me the feeling that we can run into an issue similar to https://github.com/PostHog/posthog-ios/pull/259

The other one is, if there's a constant animated screen, either just a button or really anything, does that mean that we will never capture a screenshot?

Reply:
cool, thanks for clarifying.
your call if we should check just the `rootViewController`, that can be a good start, and if we see its not enough, we can loop all its children on a next improvement, if only the `rootViewController` is enough for most cases.